### PR TITLE
Update Kubernetes to v1.28, Kubebuilder plugin to v4, operator-sdk to v1.35.0

### PR DESCRIPTION
**Pull request by halim-lee**
All tests passed:
- OCP 4.12: https://cloud.ibm.com/devops/pipelines/tekton/c3b2797d-c8ad-4dab-9660-2ec8c0643a76/runs/af863364-4083-43b4-a8b3-b5f75c6a385b/code-ci-finish/run-stage?env_id=ibm:yp:us-south
- OCP 4.16: https://cloud.ibm.com/devops/pipelines/tekton/c3b2797d-c8ad-4dab-9660-2ec8c0643a76/runs/94eb50da-4050-4f60-8295-73e255d7c258/code-ci-finish/run-stage?env_id=ibm:yp:us-south

### CODE REVIEW

**Comment by halim-lee** [comment]
Needs to be merged after pod tolerations & pod DNS policy code changes

File: cmd/main.go [hunk 1]
```diff
@@ -88,30 +91,38 @@ func main() {
 	}
 
 	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
-		Scheme:                 scheme,
-		MetricsBindAddress:     metricsAddr,
-		Port:                   9443,
+		Scheme: scheme,
+		Metrics: metricsserver.Options{
+			BindAddress: "0",
```
Comment (leochr) [inline cmd/main.go:96]: Is the metrics address changing from 8080 to 0 by this change?

File: cmd/main.go [hunk 2]
```diff
@@ -88,30 +91,38 @@ func main() {
...
+		},
+		WebhookServer: &webhook.DefaultServer{
```
Comment (leochr) [inline cmd/main.go:98]: We don't have a webhook, so wondering whether it's necessary to specify this

**Review by leochr — APPROVED** [review]
@halim-lee Thanks for the updates. Just noticed couple of things in the main.go. We can discuss tomorrow. Merging this PR to unblock other PRs.

File: cmd/main.go (hunk 2 above)
Comment (leochr) [reply cmd/main.go:98]: We confirmed that the old 'Port' field also corresponded with webhook. So decided to leave it as is (set it using the new field)

File: cmd/main.go (hunk 1 above)
Comment (leochr) [reply cmd/main.go:96]: Discussed and decided to set metricsAddr

**Pull request merged**